src/htm/regions/SPRegion.cpp

```diff
@@ -98,10 +98,10 @@ void SPRegion::initialize() {
   // algorithm requires input.
   //
   // If there are more than one input link (FAN-IN), the input buffer will be the
-  // concatination of all incomming buffers.  
+  // concatination of all incomming buffers.
   std::shared_ptr<Input> in = getInput("bottomUpIn");
   NTA_CHECK(in != nullptr);
-  if (!in->hasIncomingLinks())
+  if (!in->hasIncomingLinks() && !in->isInitialized())
```

Comment:
The intention of this check was to insure that the SP had some data to work with.
Since data can now be directly assigned to an input, we can change this to something like:

NTA_CHECK(in->getDimensions().isSpecified()) << SPRegion::initialize - No links or dimensions were set for this SP region.\n";

Reply:
Lets remove this change since we will now have a link if there is data.